catpol/spiders/cdep_career_history.py

```diff
@@ -0,0 +1,44 @@
+import scrapy
+import logging
+
+import sys
+from scrapy.http import Response
+
+import catpol.loaders as loaders
+import catpol.items as items
+import catpol.http as http
+import string
+import catpol.cmdinput as cmdinput
+from typing import Sequence, TypeVar, Coroutine
+
+
+class CdepCareerHistory(scrapy.Spider):
+    name = 'cdep_career_history'
+
+    def __init__(self, after=None, years=None):
+        super().__init__(after=None, years=None)
+        self.years = cmdinput.expand_years(after, years)
+
+    def start_requests(self) -> Coroutine:
+        for page in list(self.revert_pagination(self.years)):
+            yield http.Reqo(url=page, callback=self.get_politician_urls)
+
+    def revert_pagination(self, years: list) -> list:
+        return ['http://www.cdep.ro/pls/parlam/structura2015.de?leg={}&par=A{}'.format(year, char)
+                for year in years
+                for char in string.ascii_uppercase]
+
+    def get_politician_urls(self, response: Response) -> Coroutine:
```

Comment:
Hello Owlree,
"One way of implementing iterators is to use a restricted form of coroutine, known as a generator. By contrast with a subroutine, a generator coroutine can yield values to its caller multiple times, instead of returning just once."

"An Iterable is a simple representation of a series of elements that can be iterated over. It does not have any iteration state such as a "current element". Instead, it has one method that produces an Iterator.An Iterator is the object with iteration state. It lets you check if it has more elements using hasNext() and move to the next element (if any) using next().Typically, an Iterable should be able to produce any number of valid Iterators."

References:
https://en.wikipedia.org/wiki/Iterator
https://stackoverflow.com/questions/6863182/what-is-the-difference-between-iterator-and-iterable-and-how-to-use-them

Before putting my hands on the code, having no experience with generators, I actually learned this stuff.